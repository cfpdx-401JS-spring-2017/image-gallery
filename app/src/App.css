.App {
  text-align: center;
  display: flex;
  height: 100vh;
  width: 100%;
  justify-content: center;
  align-items: center;

}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 10vh;
  width: 20vw;
  background-color: #476A6F;
  border: none;
  margin: 2vh 2vh;
  box-shadow: 1px 1px 1px 1px #202A25;
  border-radius: 10px;
}

.Link {
  color: #F7F0F0;
  text-decoration: none;
  font-size: 2em;
}

.Link:hover {
  
}
