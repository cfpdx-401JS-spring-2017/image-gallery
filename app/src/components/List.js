import React, { Component } from 'react';
import PropTypes from 'prop-types';

class List extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data, onDelete } = this.props;

    return (
      <div>
        {data.map((img) => (
          <ul key={img._id}>
            <li className="list-title">
              {img.title}
            </li>
            <li>
              <a href="">
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
            <li>
              <button onClick={() => onDelete(img._id)}
                      className="delete-button"
              >
                Delete ðŸ—‘
              </button>
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;