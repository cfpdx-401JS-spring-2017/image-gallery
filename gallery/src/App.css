.App {
  text-align: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 8vh;
  width: 12vw;
  background-color: #2F3E46;
  color:#CAD2C5;
  border: none;
  margin: 2vh 2vh;
  box-shadow: none;
}
