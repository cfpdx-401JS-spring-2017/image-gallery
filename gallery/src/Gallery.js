import React, { Component } from 'react';
import './Gallery.css';

class Gallery extends Component {
  constructor() {
    super();
    this.state ={
        galleryView: false,
        index: 0
    };
    this.showGallery = this.showGallery.bind(this);
    this.scrollNext = this.scrollNext.bind(this);
    this.scrollLast = this.scrollLast.bind(this);
  }
  
  showGallery() {
      let { galleryView } = this.state;
      galleryView = galleryView ? false : true;
      this.setState({ galleryView: galleryView })
  }

  scrollNext(){
      let { index } = this.state;
      if (index === 2) {
          this.setState({ index: 0 });
      } else {
          index += 1;
          this.setState({ index });
      }
  }

  scrollLast(){
      let { index } = this.state;
      if (index === 0) {
          this.setState({ index: 2 });
      } else {
          index -= 1;
          this.setState({ index: index});
      }
  }

  render() {
      let { galleryView, index } = this.state;
      let {images} = this.props;
      let divvy = null;

      if (galleryView) {
         divvy = <div className="YoungDivvy" >
            <p className="exitButton" onClick={this.showGallery}>exit</p>
            <p className="arrows" onClick={this.scrollLast}> {"<"} </p>
                <div className="WhoDaNeighbors">
                    <h3 className="ImageTitle">{images[index].title}</h3>
                    <img src={images[index].url} alt=""/>
                    <h5>{images[index].description}</h5>
                </div>
            <p className="arrows" onClick={this.scrollNext}> {">"} </p>
         </div>;
      } else {
          divvy = null;
      }
    return (
      <div className="Gallery">
       <button style={{
           textDecoration: 'none',
           backgroundColor: 'white',
           color: 'black',
           border: 'solid black 5px',
           margin: '2vh',
           width: '35vw',
           height: '8vh',
           fontSize: '3em'
       }}   onClick={this.showGallery}>Go To Gallery</button>
       {divvy}
      </div>  
    );
  }
}

export default Gallery;