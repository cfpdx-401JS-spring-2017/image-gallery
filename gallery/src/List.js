import React, { Component } from 'react';

class List extends Component {
  constructor() {
    super();
    this.state ={

    };
  }
  

  render() {
    return (
      <div className="List">
        {this.props.images.map((image, i) =>{
            let {title, description, url} = image;
          return (
            <div key={i} style={{ width: '30vw', margin: '5px 5px'}}>
              <h3>{title}</h3> 
              <p>{description}</p> 
              <p>{url}</p>
            </div>
          );
          
        })}
      </div>  
    );
  }
}

export default List;