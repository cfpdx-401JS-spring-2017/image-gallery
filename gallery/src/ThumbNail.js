import React, { Component } from 'react';

class ThumbNail extends Component {
  constructor() {
    super();
    this.state ={

    };
  }
  

  render() {
    return (
      <div className="ThumbNail">
        {this.props.images.map((image, i) =>{
            let {title, url} = image;
          return (
            <div key={i} style={{
              border: '5px solid black',
              margin: '2px'
            }}>
              <h4>{title}</h4> 
              <img 
                src={url} 
                alt=""
                style={{
                    height: '25vh', 
                    width: 'auto',
                    margin: '0 5px',
                    padding: 'none'
                }}
              />
            </div>
          );
        })}
      </div>  
    );
  }
}

export default ThumbNail;