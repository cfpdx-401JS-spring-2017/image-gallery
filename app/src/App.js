import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';

import Home from './components/Home';
import Albums from './components/albums/Albums';
import './styles/App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <h2>CPDB: Cute Pomeranian Database</h2>
          <p>
            Welcome to the <code>Cute Pomeranian Database</code>! Check out our poms by clicking below:
          </p>
          <Switch>
            <Route exact path='/' component={Home}/>
            <Route path='/albums' component={Albums} />
            <Redirect to='/' />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
