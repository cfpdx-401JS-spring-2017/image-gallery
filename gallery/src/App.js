import React, { Component } from 'react';
import './App.css';
import AlbumsViewer from './AlbumsViewer';
import Albums from './Albums';

import {
  BrowserRouter as Router,
  Route, Switch, Redirect, Link
} from 'react-router-dom'

class App extends Component {
  constructor() {
    super();
    this.state = {
    }

  }

  render() {
  
    return (
      <div className="App">
        <Router>
            <div>
                <Switch>
                    {/*<Route exact path="/" component={App}/>*/}
                    <Route path="/albums/:id" component={AlbumsViewer}/>
                    <Route path="/albums" component={Albums}/>
                    <Redirect path="/albums" component={Albums}/>
                </Switch>
                <button>
                  <Link to="/albums">Go to Albums</Link>
                </button>
            </div>
        </Router>
      </div>  
    );
  }
}

export default App;
