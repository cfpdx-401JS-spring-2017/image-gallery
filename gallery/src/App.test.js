import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Gallery from './Gallery';
import ThumbNail from './ThumbNail';
import List from './List';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot of main App component', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a snapshot of Gallery component', () => {
  const bobo = renderer.create(<Gallery />).toJSON();
  expect(bobo).toMatchSnapshot();
});


