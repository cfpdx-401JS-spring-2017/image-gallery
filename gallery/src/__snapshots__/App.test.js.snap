// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="List"
  />
  <div
    className="ThumbNail"
  />
  <div
    className="Gallery"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "solid black 5px",
          "color": "black",
          "fontSize": "3em",
          "height": "8vh",
          "margin": "2vh",
          "textDecoration": "none",
          "width": "35vw",
        }
      }
    >
      Go To Gallery
    </button>
  </div>
</div>
`;

exports[`renders a snapshot of Gallery component 1`] = `
<div
  className="Gallery"
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "border": "solid black 5px",
        "color": "black",
        "fontSize": "3em",
        "height": "8vh",
        "margin": "2vh",
        "textDecoration": "none",
        "width": "35vw",
      }
    }
  >
    Go To Gallery
  </button>
</div>
`;

exports[`renders a snapshot of main App component 1`] = `
<div
  className="App"
>
  <div
    className="List"
  />
  <div
    className="ThumbNail"
  />
  <div
    className="Gallery"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "solid black 5px",
          "color": "black",
          "fontSize": "3em",
          "height": "8vh",
          "margin": "2vh",
          "textDecoration": "none",
          "width": "35vw",
        }
      }
    >
      Go To Gallery
    </button>
  </div>
</div>
`;
